pkg/api/validation/validation.go

```diff
@@ -2169,9 +2169,11 @@ func ValidatePreferredSchedulingTerms(terms []api.PreferredSchedulingTerm, fldPa
 func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopologyKey bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(podAffinityTerm.LabelSelector, fldPath.Child("matchExpressions"))...)
-	for _, name := range podAffinityTerm.Namespaces {
-		for _, msg := range ValidateNamespaceName(name, false) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), name, msg))
+	if !(len(podAffinityTerm.Namespaces) == 1 && podAffinityTerm.Namespaces[0] == "*") {
```

Comment:
Whenever regex is considered, we should think twice about scalability. Regex hides all of its computation complexity behind a simple, innocent function call. I would not use regex for checking pod affinity term which is called frequently, unless it is absolutely necessary.